src/analysis/asymptotics.lean

```diff
@@ -1061,6 +1061,96 @@ theorem is_o_iff_tendsto {f g : α → 𝕜} {l : filter α}
   is_o f g l ↔ tendsto (λ x, f x / (g x)) l (𝓝 0) :=
 iff.intro is_o.tendsto_0 (is_o_of_tendsto hgf)
 
+/-!
+### Eventually (u / v) * v = u
+
+If `u` and `v` are linked by an `is_O_with` relation, then we
+eventually have `(u / v) * v = u`, even if `v` vanishes.
+-/
+
+section eventually_mul_div_cancel
+
+variables {u v : α → 𝕜}
+
+lemma is_O_with.eventually_mul_div_cancel (h : is_O_with c u v l) :
+  (u / v) * v =ᶠ[l] u :=
+begin
+  refine eventually.mono h (λ y hy, div_mul_cancel_of_imp $ λ hv, _),
+  rw hv at *,
+  simpa using hy
+end
+
+/-! If `u = O(v)` along `l`, then `(u / v) * v = u` eventually at `l`. -/
```

Comment:
Is there a reason for this docstring choice?  Usually one uses `/-!` for section docstrings,
```suggestion
/-- If `u = O(v)` along `l`, then `(u / v) * v = u` eventually at `l`. -/
```
for lemma docstrings (same below)